snakesist/exist_client.py

```diff
@@ -225,12 +292,17 @@ def create_resource(self, collection_path: str, node: str):
         """
         Write a new document node to the database.
 
-        :param collection_path: Collection where document will be stored
+        :param collection_path: Path to collection where document will be stored,
+                                relative to the configured root collection
         :param node: XML string
         """
-        path = self._join_paths(collection_path, uuid4().hex)
-        url = f"{self.root_collection_url}/{path}"
-        self._put_request(url, node)
+        path = self._join_paths(self.root_collection, collection_path)
+        xq = (
```

Comment:
as indicated before, i prefer not to use variables that are written and read only once in a scope. but there's no common rule / practice around that.

Reply:
you're right. i tend to abuse this useless pattern. it might only make sense to use it when it improves readability drastically, and that's clearly not the case here. so i put the values of all the throw-away variables with query expressions directly into the query function.